.github/workflows/e2e-nvidia-t4-x1.yml

```diff
@@ -124,12 +124,13 @@ jobs:
           # install with Torch and build dependencies installed
           python3.11 -m pip install -v packaging wheel setuptools-scm
           python3.11 -m pip install -v .[cuda]
-        
+
       - name: Update instructlab-sdg library
         working-directory: ./sdg
         run: |
           . ../instructlab/venv/bin/activate
-          pip install .
+          pip install -v . -c constraints-dev.txt
+          pip install -v .[cuda] -c constraints-dev.txt
```

Comment:
For e2e jobs where we install `instructlab` CLI, we have to use constraints file from that other repo, not from `sdg` repo. Otherwise we install `ilab` using constraints that were not validated to work in their CI.

See an example here: https://github.com/instructlab/training/blob/473d4ee36e882ce3984873a37ce701aab0513768/.github/workflows/e2e-nvidia-l40s-x4.yml#L182-L201